js/src/oai.ts

```diff
@@ -1,157 +1,28 @@
-import { Span, startSpan } from "./logger";
-import { getCurrentUnixTimestamp } from "./util";
-
-interface ChatLike {
-  completions: any;
-}
-interface OpenAILike {
-  chat: ChatLike;
-}
+import { openAIV4NonProxyWrapper } from "./oai_wrappers/non_proxy_wrappers";
+import { openAIV4ProxyWrapper } from "./oai_wrappers/proxy_wrappers";
 
 /**
- * Wrap an `OpenAI` object (created with `new OpenAI(...)`) to add tracing. If Braintrust is
- * not configured, this is a no-op
+ * Wrap an `OpenAI` object (created with `new OpenAI(...)`) to add tracing.
  *
  * Currently, this only supports the `v4` API.
  *
- * @param openai
+ * @param openai The `OpenAI` object.
+ * @param proxyBaseUrl Use the Braintrust proxy (https://github.com/braintrustdata/braintrust-proxy) as the base URL. When left `undefined`, the URL is obtained from the environment variable `BRAINTRUST_PROXY_URL`, defaulting to `https://braintrustproxy.com/v[api_version]`. Pass `null` to leave the original base URL intact and use the non-proxy wrapper, or pass a string to use a custom URL.
```

Comment:
This seems like a fairly subtle distinction (especially between `undefined` and `null`). I think it'd be more intuitive to:

* rename `args` to `options`
* make the first argument `useProxy` and make it a `string | bool`. If `true`, it uses the default proxy url. If a string, it uses that URL. If we want to keep the proxy off by default, then maybe we create a flag like `disableProxy`.
* make the second argument `apiKey`.